src/library/psitest_library.jl

```diff
@@ -4315,3 +4129,154 @@ function build_c_sys5_radial(; raw_data, kwargs...)
     set_max_reactive_power!(load_bus3, 0.3287)
     return new_sys
 end
+
+####### Cost Function Testing Systems ################
+function _build_cost_base_test_sys(; kwargs...)
+    sys_kwargs = filter_kwargs(; kwargs...)
+    node =
+        PSY.ACBus(1, "nodeA", "REF", 0, 1.0, (min = 0.9, max = 1.05), 230, nothing, nothing)
+    load = PSY.PowerLoad("Bus1", true, node, 0.4, 0.9861, 100.0, 1.0, 2.0)
+
+    gen = ThermalStandard(;
+        name = "Cheap Unit",
+        available = true,
+        status = true,
+        bus = node,
+        active_power = 1.70,
+        reactive_power = 0.20,
+        rating = 2.2125,
+        prime_mover_type = PrimeMovers.ST,
+        fuel = ThermalFuels.COAL,
+        active_power_limits = (min = 0.0, max = 1.70),
+        reactive_power_limits = (min = -1.275, max = 1.275),
+        ramp_limits = (up = 0.02 * 2.2125, down = 0.02 * 2.2125),
+        time_limits = (up = 2.0, down = 1.0),
+        operation_cost = ThermalGenerationCost(CostCurve(LinearCurve(0.23)),
+            0.0,
+            1.5,
+            0.75,
+        ),
+        base_power = 100.0,
+    )
+
+    DA_load_forecast = SortedDict{Dates.DateTime, TimeSeries.TimeArray}()
+    ini_time = DateTime("1/1/2024  0:00:00", "d/m/y  H:M:S")
+    # Load levels to catch each segment in the curves
+    load_forecasts = [[2.1, 3.4, 2.76, 3.0, 1.0], [1.3, 3.0, 2.1, 1.0, 1.0]]
+    for (ix, date) in enumerate(range(ini_time; length = 2, step = Hour(1)))
+        DA_load_forecast[date] =
+            TimeSeries.TimeArray(
+                range(ini_time; length = 5, step = Hour(1)),
+                load_forecasts[ix],
+            )
+    end
+    load_forecast = PSY.Deterministic("max_active_power", DA_load_forecast)
+    cost_test_sys = PSY.System(100.0;)
+    PSY.add_component!(cost_test_sys, node)
+    PSY.add_component!(cost_test_sys, load)
+    PSY.add_component!(cost_test_sys, gen)
+    PSY.add_time_series!(cost_test_sys, load, load_forecast)
+    return cost_test_sys
+end
+
+function build_linear_cost_test_sys(; kwargs...)
+    base_sys = _build_cost_base_test_sys(; kwargs...)
+    node = PSY.get_component(ACBus, base_sys, "nodeA")
+    test_gen = thermal_generator_linear_cost(node)
+    PSY.add_component!(base_sys, test_gen)
+    return base_sys
+end
+
+function build_linear_fuel_test_sys(; kwargs...)
```

Comment:
Nitpick: could eliminate duplication by defining a helper function
```julia
function build_cost_test_sys(gen_func; kwargs...)
    ...
    test_gen = gen_func(node)
    ...
end
```
(or fold this into `_build_cost_base_test_sys`) and then each of these is like
```suggestion
build_linear_fuel_test_sys(; kwargs...) =
    build_cost_test_sys(thermal_generator_linear_fuel; kwargs...)
```